### DISCUSSION TITLE
20220525

### DISCUSSION THREAD

**Discussion by tmiw** [ARSFI/paclink-open-source#24] [General]
This is a pre-production build intended to gauge current status of the project and determine what needs to be fixed. Brief testing indicates that packet (using Kenwood TM-D700A), AGW (using AGWPE and UZ7HO soundmodem) and Telnet channels work; Pactor is TBD.

Contains (vs. [last pre-release version](https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220515)):

* Installer: Paclink help file now included.
* Pactor: Fix issue preventing Connect dialog box from displaying correct dial frequency.
* Pactor: Fix cross-threading issue that is possibly contributing to crashes.

Instructions:

1. Unzip paclink-open-source.zip somewhere on your hard drive.
2. Run setup.exe to install on your system. Paclink will be installed in the Start menu.
3. Open Paclink from the Start menu and reconfigure if required. (There is no import functionality from the existing Paclink at this time but if you used a pre-release version, configuration should be saved.)

Please leave comments in the release's discussion thread or email K6AQ@winlink.org with feedback. 👍 

<hr /><em>This discussion was created from the release <a href='https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220525'>20220525</a>.</em>

**Comment by kd4vrf** [comment]
I've tested the help by clicking "Help --> Contents..." and I still don't see a help window. I get the blue spinning disk and then nothing.

The Dial frequency displayed as expected for me on this build.

I haven't seen any crashes yet today I'll keep testing it to see if any crashes occur and let you know.

I do have another issue I have noticed, this issue happened on one of the previous builds and again today on the new build.

Do you prefer that I just list the details here in the thread, or should I create an issue and put the details there?

**Reply by tmiw** [reply]
I'll look into the help thing some more. Does the Paclink.chm render properly if you double click on it at least? It should be in C:\Program Files (x86)\Winlink\Paclink.

As for the other issue, you can describe it here. 👍

**Reply by kd4vrf** [reply]
It does render properly when I open it directly. I clicked around for a bit and all of the help pages in the contents loaded just fine.

I'll pull the logs and type up the description of the issue shortly.

**Comment by kd4vrf** [comment]
When I check into the "Winlink Wednesday" Net I'll get a confirmation email of my check in and an email letting me know the roster has been published. Last week when I downloaded those messages from Paclink they disappeared. Paclink showed in the window that it was retrieving the messages but when I synced my email client (Microsoft Mail) there were no messages to retrieve.

Last week I sent a bunch of messages to myself after that and it never happened again so I thought it was a fluke. This week I paid attention when Paclink was retrieving those messages. The "To Clients:" at the bottom of the Paclink window shows 13 messages. When Paclink was done downloading the Winlink Wednesday emails that count didn't change and stayed at 13.

Downloading messages:
![Screen Shot 2022-05-26 at 2 53 28 PM](https://user-images.githubusercontent.com/45251519/170606282-1e351b15-9589-460c-bda7-f9a875de615b.png)
 
Completed:
![Screen Shot 2022-05-26 at 2 54 15 PM](https://user-images.githubusercontent.com/45251519/170606518-4b447749-29e5-47f5-a61e-bfd1f5178a40.png)

I don't receive any errors, everything appears to work as expected. However, there aren't any messages to retrieve when I sync my mail client with Paclink.

This is exactly what happened last week with exactly those same messages. The only thing I can think of that is different with those particular messages is that the to field has all of the net participants, that is 400+ winlink addresses in the "to:" field on each email. Other than that the messages are unremarkable. They don't contain any attachments and are short text only messages.

Here are the Logs:

Paclink Log
[Paclink.log](https://github.com/ARSFI/paclink-open-source/files/8783101/Paclink.log)

Compressed Error Logs:
[Logs.zip](https://github.com/ARSFI/paclink-open-source/files/8783111/Logs.zip)

**Reply by tmiw** [reply]
Thanks for the info. I was able to duplicate the issue with only two email addresses in the To field; turns out there was an off by one indexing issue that prevented the email addresses from properly parsing and thus prevented storage of the message. This (and the help issue) should now be resolved in the latest build: https://github.com/ARSFI/paclink-open-source/releases/tag/nightly_20220526.

Anyway, let me know whether you still run into either issue.
